web/src/avatar.ts

```diff
@@ -57,21 +58,23 @@ export function build_bot_edit_widget($target) {
     );
 }
 
-function display_avatar_delete_complete() {
+function display_avatar_delete_complete(): void {
     $("#user-avatar-upload-widget .upload-spinner-background").css({visibility: "hidden"});
     $("#user-avatar-upload-widget .image-upload-text").show();
     $("#user-avatar-source").show();
 }
 
-function display_avatar_delete_started() {
+function display_avatar_delete_started(): void {
     $("#user-avatar-upload-widget .upload-spinner-background").css({visibility: "visible"});
     $("#user-avatar-upload-widget .image-upload-text").hide();
     $("#user-avatar-upload-widget .image-delete-button").hide();
 }
 
-export function build_user_avatar_widget(upload_function) {
-    const get_file_input = function () {
-        return $("#user-avatar-upload-widget .image_file_input").expectOne();
+export function build_user_avatar_widget(upload_function: UploadFunction): void {
+    const get_file_input = function (): JQuery<HTMLInputElement> {
+        return $(
+            "#user-avatar-upload-widget .image_file_input",
+        ).expectOne() as JQuery<HTMLInputElement>;
```

Comment:
Perhaps we can fix our definition of `expectOne` on `JQuery` to get this working without the cast